external/urlfrontier/src/main/java/com/digitalpebble/stormcrawler/urlfrontier/StatusUpdaterBolt.java

```diff
@@ -170,16 +175,27 @@ public void prepare(
         }
 
         if (address == null) {
-            channel =
-                    ManagedChannelUtil.createChannel(
-                            ConfUtils.getString(
-                                    stormConf, URLFRONTIER_HOST_KEY, URLFRONTIER_DEFAULT_HOST),
-                            ConfUtils.getInt(
-                                    stormConf, URLFRONTIER_PORT_KEY, URLFRONTIER_DEFAULT_PORT));
-        } else {
-            channel = ManagedChannelUtil.createChannel(address);
+            String host =
+                    ConfUtils.getString(stormConf, URLFRONTIER_HOST_KEY, URLFRONTIER_DEFAULT_HOST);
+            int port = ConfUtils.getInt(stormConf, URLFRONTIER_PORT_KEY, URLFRONTIER_DEFAULT_PORT);
+            address = host + ":" + port;
         }
 
+        initFrontierConnection();
+
+        // Time interval in seconds for checking the connection to the URL Frontier in case of
+        // erroneous connection
+        int connectionCheckerInterval =
+                ConfUtils.getInt(stormConf, URLFRONTIER_CONNECTION_CHECKER_INTERVAL, 60);
+
+        // Create a periodic ExecutorService to check if the connection to the URLFrontier is
+        // still alive. If not, try to reconnect.
+        ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
+        executorService.scheduleAtFixedRate(this, 1, connectionCheckerInterval, TimeUnit.SECONDS);
+    }
+
+    public void initFrontierConnection() {
+        channel = ManagedChannelUtil.createChannel(address);
```

Comment:
Do we really need to re-create the managed channel? 

I was under the impression, that `io.grpc.ManagedChannel` is capable of handling a re-connect after a connection loss or error? (as stated in https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md or discussed in https://groups.google.com/g/grpc-io/c/P1Qr6e0I1yA )

Reply:
Same, this was my assumption

Reply:
Thank you very much for your comments. I worked it over and omitted the re-creation of a new channel. Initially, I thought it would be necessary as it is hard to grasp when the automatic re-connection takes place and it leaded to some test scenarios, when for no obvious reason sometimes it worked and sometimes it didn't, depending on whether the channel has already been automatically re-connected or not